### ISSUE TITLE
Parse new search

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#4977]
From IRC:

```
22:15:51bananeweizen: Is it just me, or is this relatively new: https://www.geocaching.com/play/search
22:16:06bananeweizen: because it seems that the new search is working okay now
22:16:16bananeweizen: which means we should prepare for parsing it
22:16:36bananeweizen: and afterwards we could also provide "search templates" like shown here
```

I have to admit I did not use/test it up to now.
Last time I checked it briefly the result page contained much less information.

**Comment by Lineflyer** [comment]
Still open and old search is probably still working as by users demand and us notifying Groundspeak about it (in 2018).

**Comment by fm-sys** [comment]
The API used by the new search page is this one:
```
https://www.geocaching.com/play/search?ot=4&sizes=2,8,3,6&cc=1&utr=false
```
In this case, I searched for Geocaches which are not `Large` and have updated coordinates. All in all this looks quite easy to decode...

---
Note that this search API is much more advanced than the one used for the live map, but does not contain coordinates. Anyway, it would be pretty cool, if this whole thing could be integrated into c:geo!

![grafik](https://user-images.githubusercontent.com/64581222/117864986-b3b84000-b295-11eb-9753-a12617ed1bd3.png)

**Comment by eddiemuc** [comment]
Have a little patience :-)

**Comment by fm-sys** [comment]
No problem, just wanted to keep things organized. I haven't expect anything to happen here soon, and we have currently way more urgent things to do. And I really admire your excellent work for cgeo, you have already solved so many long overdue issues! :)

**Comment by eddiemuc** [comment]
My current rough idea to implement this as a first step (!) is this:
* enable connectors (starting with gc.com) to take a filter object for search
* In the search dialog, add a new Option "Search by filter"
* Brings you to the filter dialog, upon "go" the search is done and result displayed in cache list as today

To make this possible, in a step 0 I have to switch search logic under the hood from "old" gc.com search to "WebApi" - otherwise filter capabilities of gc.com would not be available to use.

Several questions though:
* Todays search limit of 20 is a result of parsing the old search pages. It is no longer necessary when using WebApi. Should the limit be raised (e.g. to 50 or 100)?
* Not all filters can be performed by gc.com WebAPI. Also "inverse" and "or" connections are not supported. I would thus implement "search by filter" such that it tries to apply filter as best as possible, remaining filters would then be applied on the retrieved result (this might then lead to concrete result count showing to user being less than e.g. 20/50/100 - as it is today). Ok?
* Now a tricky one: If in the search result list the user **changes** the filter: should we requery the servers in this case or should we apply the changed filtering only on the downloaded caches? (for offline cache filtering, which is SQL-based now, database is requeried in this case)

@fm-sys ?

**Comment by fm-sys** [comment]
> Todays search limit of 20 is a result of parsing the old search pages. It is no longer necessary when using WebApi. Should the limit be raised (e.g. to 50 or 100)?

I think this would indeed makes sense, especially if the user uses complex filters which need to be partly executed in-app. 

> I would thus implement "search by filter" such that it tries to apply filter as best as possible, remaining filters would then be applied on the retrieved result (this might then lead to concrete result count showing to user being less than e.g. 20/50/100 - as it is today). Ok?

Yes, fully agree 👍

> Now a tricky one: If in the search result list the user changes the filter: should we requery the servers in this case or should we apply the changed filtering only on the downloaded caches?

Only filtering the offline results would maybe be confusing if the user explicitly wants to do an online search. So probably a requery would be the best idea...

**Comment by jakedot** [comment]
> Only filtering the offline results would maybe be confusing if the user explicitly wants to do an online search. So probably a requery would be the best idea...

Maybe we should distinguish this behaviour from the current approach of (displayed list) -> filter -> (filtered list) -> changed filter -> (filtered list by applying both filters)

Further filtering would of course be simple with an "and" filter, but it is different from the old behaviour, maybe this would need to be highlighted?

**Comment by eddiemuc** [comment]
Solved with PR #10622 . Be aware of limitations as mentioned in #10633 and #10621. Search limit is no longer just 20.
